src/module-loader/dotnet/src/MorganStanley.ComposeUI.ModuleLoader.Abstractions/LifetimeEvent.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Morgan Stanley makes this available to you under the Apache License,
+// Version 2.0 (the "License"). You may obtain a copy of the License at
+// 
+//      http://www.apache.org/licenses/LICENSE-2.0.
+// 
+// See the NOTICE file distributed with this work for additional information
+// regarding copyright ownership. Unless required by applicable law or agreed
+// to in writing, software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions
+// and limitations under the License.
+
+namespace MorganStanley.ComposeUI.ModuleLoader;
+
+public sealed class LifetimeEvent
+{
+    private LifetimeEvent(Guid instanceId, string eventType, bool isExpected)
```

Comment:
Are there lifetime events that are not known at compile time? How about adding an enumeration instead?

```cs
public enum LifetimeEventType
{
    Starting,
    Started,
    Stopping,
    Stopped
}
```

Reply:
I vaguely remember that we added LifetimeEventTypes as strings and not enums because of the JavaScript counterparts

Reply:
Isn't that true for any enum? They can be de/serialized from strings. The performance impact is negligible, but strings don't provide the same type safety (eg. subscribing to the wrong event type is possible).